### PR TITLE
test for add step

### PR DESCRIPTION
**Pull request by keshav234156**
Fixes #1242

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `npm test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts
* [ ] PR is descriptively titled
* [ ] ask `@publiclab/is-reviewers` for help, in a comment below
* [ ] Insert-step functionality is working correct as expected.
> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software
Please make sure to get at least two reviews before asking for merging the PR as that would make the PR more reliable on our part
Thanks!

### CODE REVIEW

**Comment by keshav234156** [comment]
cc: @jywarren  @HarshKhandeparkar

File: src/test/frontend.test.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+const timeout = process.env.SLOWMO ? 30000 : 10000;
+const fs = require('fs');
+beforeAll(async () => {
+
+  await page.goto('file:///home/keshav/image-sequencer/examples/index.html', {waitUntil: 'domcontentloaded'});
```
Comment (keshav234156) [inline src/test/frontend.test.js]: Please change the path here.I am working on it to make it relative path as suggested by @HarshKhandeparkar

**Comment by keshav234156** [comment]
run it with npm run test-ui-2

**Comment by jywarren** [comment]
This is awesome. Following eagerly! Please ping me once the path is local. Great work! cc @HarshKhandeparkar @vaibhavmatta @tech4gt and others! ♥️😁🎉

**Comment by jywarren** [comment]
Could you add the run command to the overall test trigger so they run in Travis?

**Comment by keshav234156** [comment]
@jywarren Do you know how can we make relative to examples/index.html ?

**Comment by keshav234156** [comment]
ohk done with fs library as suggested by @HarshKhandeparkar

**Comment by keshav234156** [comment]
cc: @jywarren
Screenshot from 2019-12-18 04-21-19

https://travis-ci.org/publiclab/image-sequencer/jobs/626434689#L1683


Horray!!!
We have now test running on Travis as well

**Comment by keshav234156** [comment]
@jywarren Now we just need to add it to npm test-ui command properly . How can we do that?

**Comment by jywarren** [comment]
This is Awesome!!! Can you try breaking a test to be sure it triggers a Travis failure? 🎉🎉🎉

File: package.json [hunk 2]
```diff
@@ -7,6 +7,7 @@
     "debug": "TEST=true node ./index.js -i ./examples/images/monarch.png -s invert",
     "test": "TEST=true istanbul cover tape test/core/*.js test/core/ui/user-interface.js test/core/modules/*.js | tap-spec; node test/core/sequencer/benchmark.js;  browserify test/core/sequencer/meta-modules.js test/core/sequencer/image-sequencer.js test/core/sequencer/chain.js  test/core/sequencer/replace.js test/core/sequencer/import-export.js test/core/sequencer/run.js test/core/sequencer/dynamic-imports.js test/core/util/*.js | tape-run --render=\"tap-spec\"",
     "test-ui": "node node_modules/jasmine/bin/jasmine test/ui/spec/*.js",
+    "test-ui":"node ./node_modules/.bin/jest",
```
Comment (jywarren) [inline package.json]: I think it has to be a unique name!

**Comment by keshav234156** [comment]
let's check with title

File: package.json (hunk 2 above)
Comment (keshav234156) [reply package.json]: yes,but just for the sake that it runs on travis i have done this.How can we add command to travis?

**Comment by keshav234156** [comment]
@jywarren travis failed as i changed title of page

File: src/test/frontend.test.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        
+  }, timeout);
+});
+describe('Add step', () => {
+  test('length is increased', async () => {
+    	await page.waitForSelector('.step');
+    const previousLength = await page.evaluate(() => document.querySelectorAll('.step').length);
+    await page.click('[data-value=\'brightness\']');
+    const previousLength1 = await page.evaluate(() => document.querySelectorAll('.step').length);
+    console.log(previousLength);
```
Comment (jywarren) [inline src/test/frontend.test.js]: Ok awesome!!!! Let's remove some comments and fix the test again and then let's merge this!!!

**Comment by jywarren** [comment]
Let's just remove a few of the console.log statements too and add a brief readme entry? Thanks, this is amazing!!!!!

**Comment by keshav234156** [comment]
@jywarren LGTM

**Comment by jywarren** [comment]
Ah, I see a merge conflict, can you rebase? Then good to go!!!!! Thanks!

**Comment by keshav234156** [comment]
in which file?

**Comment by jywarren** [comment]
Oh sorry! Take a look - GitHub has marked the pr as having a conflict. It
should show which! But, you can rebase the whole branch to resolve it.

**Comment by keshav234156** [comment]
@jywarren where it is shown in Github whether Pr is having conflicts

**Comment by jywarren** [comment]
Ah! I dunno what happened, I saw a message earlier but now it's just asking for 2 reviews. I can bypass but let's aim for 2 in future PRs! Thanks!

**Comment by jywarren** [comment]
🎉🎉🎉🎉 awesome. Would you have interest in the list of tests issue and/or the readme update? This is so great!!!!!

**Comment by keshav234156** [comment]
Ok will make list of ui test nedded.some of them are already listed in #1000

**Pull request merged**